### DISCUSSION TITLE
Confusing mididevice and midiconfig .conf explanations

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#4031] [General]
![image](https://github.com/user-attachments/assets/ba9bfdb7-0e50-40a5-95ec-c51ea204984e)

As you can see per .conf:
- mididevice: Set where MIDI data from the emulated MPU-401 MIDI interface is sent
  - auto:        Either one of the built-in MIDI synthesisers (if `midiconfig` is set to 'fluidsynth' or 'mt32')

I did `mididevice=auto`, `midiconfig=mt32`, but mt32 didn't get activated. That kind of matches _another_ .conf description:
- midiconfig: Configuration options for the selected MIDI interface (unset by default).
  - This **option has no effect** when using the built-in synthesizers ('mididevice = fluidsynth' or 'mididevice = mt32').

mt32 activates when entered at `mididevice=mt32`:
![image](https://github.com/user-attachments/assets/692ded75-d107-4a71-bd6d-d7b42cb7f0c6)

So, it seems `mididevice=auto` either behaves wrongly or its description is wrong?

**Comment by johnnovak** [comment]
@Torinde I checked earlier releases, the code, and the history, and I don't think this ever worked in Staging 😆 LOL 🤣 So yeah, the description is lying...

The whole `mididevice = auto` combined with `midiconfig = mt32 | fluidsynth` is dodgy, confusing, and completely unnecessary. I'll get rid of it, along with some further simplifications.

**Reply by Torinde** [reply]
To be fixed in #4040
